docs/guide/interchange/05-creating-sell-orders.md

```diff
@@ -0,0 +1,687 @@
+func (b *BuyOrderBook) FillSellOrder(order Order) (remainingSellOrder Order, liquidated []Order, gain int32, filled bool) {
+	var liquidatedList []Order
+	totalGain := int32(0)
+	remainingSellOrder = order
+	// Liquidate as long as there is match
+	for {
+		var match bool
+		var liquidation Order
+		remainingSellOrder, liquidation, gain, match, filled = b.LiquidateFromSellOrder(
+			remainingSellOrder,
+		)
+		if !match {
+			break
+		}
+		// Update gains
+		totalGain += gain
+		// Update liquidated
+		liquidatedList = append(liquidatedList, liquidation)
+		if filled {
+			break
+		}
+	}
+	return remainingSellOrder, liquidatedList, totalGain, filled
+}
+```
+
+#### `LiquidateFromSellOrder`
+
+`LiquidateFromSellOrder` liquidates the first buy order of the book from the sell order if no match is found, return false for match.
+
+```go
+// x/ibcdex/types/buy_order_book.go
+func (b *BuyOrderBook) LiquidateFromSellOrder(order Order) ( remainingSellOrder Order, liquidatedBuyOrder Order, gain int32, match bool, filled bool) {
+  remainingSellOrder = order
+  // No match if no order
+  orderCount := len(b.Book.Orders)
+  if orderCount == 0 {
+    return order, liquidatedBuyOrder, gain, false, false
+  }
+  // Check if match
+  highestBid := b.Book.Orders[orderCount-1]
+  if order.Price > highestBid.Price {
+    return order, liquidatedBuyOrder, gain, false, false
+  }
+  liquidatedBuyOrder = *highestBid
+  // Check if sell order can be entirely filled
+  if highestBid.Amount >= order.Amount {
+    remainingSellOrder.Amount = 0
+    liquidatedBuyOrder.Amount = order.Amount
+    gain = order.Amount * highestBid.Price
+    // Remove highest bid if it has been entirely liquidated
+    highestBid.Amount -= order.Amount
+    if highestBid.Amount == 0 {
+      b.Book.Orders = b.Book.Orders[:orderCount-1]
+    } else {
+      b.Book.Orders[orderCount-1] = highestBid
+    }
+    return remainingSellOrder, liquidatedBuyOrder, gain, true, true
+  }
+  // Not entirely filled
+  gain = highestBid.Amount * highestBid.Price
+  b.Book.Orders = b.Book.Orders[:orderCount-1]
+  remainingSellOrder.Amount -= highestBid.Amount
+  return remainingSellOrder, liquidatedBuyOrder, gain, true, false
+}
+```
+
+### `OriginalDenom`
+
+`OriginalDenom` returns back the original denom of the voucher. False is returned if the port ID and channel ID provided are not the origins of the voucher
+
+```go
+// x/ibcdex/keeper/denom.go
+func (k Keeper) OriginalDenom(ctx sdk.Context, port string, channel string, voucher string) (string, bool) {
+	trace, exist := k.GetDenomTrace(ctx, voucher)
+	if exist {
+		// Check if original port and channel
+		if trace.Port == port && trace.Channel == channel {
+			return trace.Origin, true
+		}
+	}
+	// Not the original chain
+	return "", false
+}
+```
+
+If a token is an IBC token (has an `ibc/` prefix) `SafeMint` mints IBC tokens with `MintTokens`, otherwise, it unlocks native tokens with `UnlockTokens`.
+
+### `SafeMint`
+
```

Comment:
```suggestion
If a token is an IBC token (has an `ibc/` prefix) `SafeMint` mints IBC tokens with `MintTokens`, otherwise, it unlocks native tokens with `UnlockTokens`.
```